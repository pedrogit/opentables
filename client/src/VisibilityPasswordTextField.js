import React from "react";
import { TextField, InputAdornment, IconButton } from "@mui/material";
import VisibilityIcon from "@mui/icons-material/Visibility";
import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";

function VisibilityPasswordTextField({ ...rest }) {
  //const { isVisible, onVisibilityChange,  = props;
  const [isVisible, toggleVisible] = React.useState(false);

  return (
    <TextField
      {...rest}
      type={isVisible ? "text" : "password"}
      InputProps={{
        endAdornment: (
          <InputAdornment position="end">
            <IconButton
              aria-label="Toggle password visibility"
              onClick={() => {
                toggleVisible(!isVisible);
              }}
              onMouseDown={(event) => {
                event.preventDefault();
              }}
            >
              {isVisible ? <VisibilityIcon /> : <VisibilityOffIcon />}
            </IconButton>
          </InputAdornment>
        ),
      }}
    ></TextField>
  );
}

export default VisibilityPasswordTextField;
