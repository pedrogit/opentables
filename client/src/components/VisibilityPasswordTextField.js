import React from "react";
import { TextField, InputAdornment, IconButton } from "@mui/material";
import VisibilityIcon from "@mui/icons-material/Visibility";
import VisibilityOffIcon from "@mui/icons-material/VisibilityOff";
import Tooltip from "@mui/material/Tooltip";

function VisibilityPasswordTextField({ ...rest }) {
  const [isVisible, toggleVisible] = React.useState(false);

  const defaultSx = { backgroundColor: "white", paddingRight: "0px" };

  const {
    name,
    value,
    label,
    required,
    sx,
    inputProps,
    variant,
    size,
    fullWidth,
    autoComplete,
    inputRef,
    onChange,
    onKeyDown,
    error,
    InputLabelProps,
  } = rest;
  return (
    <TextField
      name={name}
      value={value}
      label={label}
      required={required}
      sx={sx}
      inputProps={inputProps}
      variant={variant}
      size={size}
      type={isVisible ? "text" : "password"}
      fullWidth={fullWidth}
      autoComplete={autoComplete}
      onChange={onChange}
      onKeyDown={onKeyDown}
      error={error}
      inputRef={inputRef}
      InputLabelProps={InputLabelProps}
      // eslint-disable-next-line react/jsx-no-duplicate-props
      InputProps={{
        id: "passwordInput",
        sx: { ...defaultSx, ...rest.sx, marginTop: "0px", marginBottom: "0px" },
        endAdornment: (
          <InputAdornment position="end">
            <Tooltip title="View Password">
              <IconButton
                aria-label="Toggle password visibility"
                onClick={() => {
                  toggleVisible(!isVisible);
                }}
                onMouseDown={(event) => {
                  event.preventDefault();
                }}
              >
                {isVisible ? <VisibilityIcon /> : <VisibilityOffIcon />}
              </IconButton>
            </Tooltip>
          </InputAdornment>
        ),
      }}
    />
  );
}

export default VisibilityPasswordTextField;
