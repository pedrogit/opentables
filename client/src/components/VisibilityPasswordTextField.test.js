import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import VisibilityPasswordTextField from './VisibilityPasswordTextField';

describe('VisibilityPasswordTextField', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<VisibilityPasswordTextField />, div);
  });

  it('has a valid snapshot', () => {
    const component = renderer.create(
      <VisibilityPasswordTextField />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});