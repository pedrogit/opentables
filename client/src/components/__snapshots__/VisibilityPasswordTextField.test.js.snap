// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisibilityPasswordTextField has a valid snapshot 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
>
  <div
    className="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedEnd css-o9k5xi-MuiInputBase-root-MuiOutlinedInput-root"
    onClick={[Function]}
  >
    <input
      aria-invalid={false}
      autoFocus={false}
      className="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
      disabled={false}
      onAnimationStart={[Function]}
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      required={false}
      type="password"
    />
    <div
      className="MuiInputAdornment-root MuiInputAdornment-positionEnd MuiInputAdornment-outlined MuiInputAdornment-sizeMedium css-1laqsz7-MuiInputAdornment-root"
    >
      <button
        aria-label="Toggle password visibility"
        className="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-78trlr-MuiButtonBase-root-MuiIconButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
          data-testid="VisibilityOffIcon"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
          />
        </svg>
      </button>
    </div>
    <fieldset
      aria-hidden={true}
      className="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="css-nnbavb"
      >
        <span
          className="notranslate"
          dangerouslySetInnerHTML={
            Object {
              "__html": "&#8203;",
            }
          }
        />
      </legend>
    </fieldset>
  </div>
</div>
`;
